src/foam/nanos/notification/Notification.js

```diff
@@ -114,5 +120,49 @@ foam.CLASS({
       name: 'slackMessage',
       documentation: 'Message to be sent to Slack if sendSlackMessage is enabled.'
     }
+  ],
+
+  methods: [
+    {
+      name: 'checkOwnership',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      type: 'Boolean',
+      javaCode: `
+        foam.nanos.auth.User user = (foam.nanos.auth.User) x.get("user");
+        if( user == null ) return false;
+
+        return getUserId() == user.getId();
+      `
+    },
+    {
+      name: 'authorizeOnCreate',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! auth.check(x, "notificationDAO.read.*") ) throw new AuthorizationException("You don't have permission to create this notification.");
+      `
+    },
+    {
+      name: 'authorizeOnUpdate',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! auth.check(x, "notificationDAO.update.*") ) throw new AuthorizationException("You don't have permission to update notifications you do not own.");
+      `
+    },
+    {
+      name: 'authorizeOnDelete',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! auth.check(x, "*") ) throw new AuthorizationException("You don't have permission to delete notifications you do not own.");
+      `
+    },
+    {
+      name: 'authorizeOnRead',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! checkOwnership(x) && ! auth.check(x, "notificationDAO.read.*") ) throw new AuthorizationException("You don't have permission to read notifications you do not own.");
```

Comment:
@penzital @noodlemoodle Is authorizeOnRead called on every object put in the sink on select()?  Curious how checkOwnership works?

Reply:
Yes, the authorizer containing this is passed into the authorizationDAO which adds HAS_PERMISSION as part of the predicate in select. HAS_PERMISSION calls the authorizeOnRead on every object before putting it into the sink I believe. this predicate was added in this pr https://github.com/foam-framework/foam2/pull/2395
so in this case a read on a notification would be granted if the notification belong to current user or current user has admin permission.

Reply:
authorizeOnRead is called on every object unless current user has global read permission